### PR TITLE
OpenAPI spec for spend overview summary API

### PR DESCRIPTION
**Pull request by sanskarsharma**

### CODE REVIEW

File: src/components/parameters/card_ids.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+in: query
+name: card_ids
```
Comment (snarayanank2) [inline src/components/parameters/card_ids.yaml:2]: why do you need card_ids plural?

File: src/admin/paths/admin@spend_overview@summary.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    - $ref: ../../components/parameters/time_interval_start.yaml
+    - $ref: ../../components/parameters/time_interval_end.yaml
+    # - $ref: ../../components/parameters/is_card_spend.yaml  # will un-comment later when this filter is available
+    - $ref: ../../components/parameters/is_reimbursable.yaml
+    - $ref: ../../components/parameters/merchants.yaml
+    - $ref: ../../components/parameters/category_ids.yaml
+    - $ref: ../../components/parameters/project_ids.yaml
+    - $ref: ../../components/parameters/department_ids.yaml
+    - $ref: ../../components/parameters/user_ids.yaml
+    # - $ref: ../../components/parameters/card_ids.yaml  # will un-comment later when this filter is available
```
Comment (snarayanank2) [inline src/admin/paths/admin@spend_overview@summary.yaml:21]: why do you need card_ids plural?

File: src/components/parameters/category_ids.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+in: query
+name: category_ids
```
Comment (snarayanank2) [inline src/components/parameters/category_ids.yaml:2]: why is this plural?

File: src/admin/paths/admin@spend_overview@summary.yaml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    This API supports very rich filtering on all response fields and pagination via query parameters. To understand how to use these parameters, please see the [Guide to Data APIs](https://docs.fylehq.com/docs/fyle-platform-docs/ZG9jOjI3Mzk2NTM3-guide-to-data-ap-is#filtering)
+  parameters:
+    - $ref: ../../components/parameters/time_interval_granularity.yaml
+    - $ref: ../../components/parameters/time_interval_preset.yaml
+    - $ref: ../../components/parameters/time_interval_start.yaml
+    - $ref: ../../components/parameters/time_interval_end.yaml
+    # - $ref: ../../components/parameters/is_card_spend.yaml  # will un-comment later when this filter is available
+    - $ref: ../../components/parameters/is_reimbursable.yaml
+    - $ref: ../../components/parameters/merchants.yaml
+    - $ref: ../../components/parameters/category_ids.yaml
```
Comment (snarayanank2) [inline src/admin/paths/admin@spend_overview@summary.yaml:17]: why are these parameters plural? they should all be singular right?

client will be sending `category_id=in.(c1, c2)`

File: src/components/parameters/department_ids.yaml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+in: query
+name: department_ids
```
Comment (snarayanank2) [inline src/components/parameters/department_ids.yaml:2]: singular

File: src/components/parameters/time_interval_preset.yaml [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+  Supported operator is `eq`.
+schema:
+  type: string
+  default: this_year
+  enum:
+    - this_week
+    - this_month
+    - this_quarter
+    - this_year
+  example: eq.this_year
```
Comment (snarayanank2) [inline src/components/parameters/time_interval_preset.yaml:16]: example is not matching the enum above

File: src/components/parameters/time_interval_granularity.yaml [hunk 7]
```diff
@@ -0,0 +1,15 @@
+
+in: query
+name: time_interval_granularity
+description: |
+  The time duration by which to group expenses to aggregate sum of amount.
+  Supported operator is `eq`.
+schema:
+  type: string
+  default: month
+  enum:
```
Comment (snarayanank2) [inline src/components/parameters/time_interval_granularity.yaml:10]: example is not matching enum

File: src/components/parameters/user_ids.yaml [hunk 8]
```diff
@@ -0,0 +1,8 @@
+in: query
+name: user_ids
```
Comment (snarayanank2) [inline src/components/parameters/user_ids.yaml:2]: singular

**Review by snarayanank2 — CHANGES_REQUESTED** [review]

File: src/components/parameters/card_ids.yaml (hunk 1 above)
Comment (sanskarsharma) [reply src/components/parameters/card_ids.yaml:2]: its a query param for filtering. Say in spend by merchant tab admin wants to see spend from specific cards

In design, few filterns supported single value and few supported multiple values. So I am supporting multiple values for all in API. Web-app can still send single value in a list

**Pull request merged**